src/box2dx/Box2D.NetStandard/Dynamics/Contacts/ContactSolver.cs

```diff
@@ -39,779 +39,771 @@
+
+							vB += mB * (P1 + P2);
+							wB += iB * (Vectex.Cross(cp1.rB, P1) + Vectex.Cross(cp2.rB, P2));
+
+							// Accumulate
+							cp1.normalImpulse = x.X;
+							cp2.normalImpulse = x.Y;
+
+							break;
+						}
+
+						//
+						// Case 4: x1 = 0 and x2 = 0
+						// 
+						// vn1 = b1
+						// vn2 = b2;
+						x.X = 0.0f;
+						x.Y = 0.0f;
+						vn1 = b.X;
+						vn2 = b.Y;
+
+						if (vn1 >= 0.0f && vn2 >= 0.0f)
+						{
+							// Resubstitute for the incremental impulse
+							Vector2 d = x - a;
+
+							// Apply incremental impulse
+							Vector2 P1 = d.X * normal;
+							Vector2 P2 = d.Y * normal;
+							vA -= mA * (P1 + P2);
+							wA -= iA * (Vectex.Cross(cp1.rA, P1) + Vectex.Cross(cp2.rA, P2));
+
+							vB += mB * (P1 + P2);
+							wB += iB * (Vectex.Cross(cp1.rB, P1) + Vectex.Cross(cp2.rB, P2));
+
+							// Accumulate
+							cp1.normalImpulse = x.X;
+							cp2.normalImpulse = x.Y;
+						}
+
+						// No solution, give up. This is hit sometimes, but it doesn't seem to matter.
+						break;
+					}
+				}
+
+				_velocities[indexA].v = vA;
+				_velocities[indexA].w = wA;
+				_velocities[indexB].v = vB;
+				_velocities[indexB].w = wB;
+			}
+		}
+
+		public void StoreImpulses()
+		{
+			for (var i = 0; i < _count; ++i)
+			{
+				ContactVelocityConstraint vc = _velocityConstraints[i];
+				Manifold manifold = _contacts[vc.contactIndex].Manifold;
+
+				for (var j = 0; j < vc.pointCount; ++j)
+				{
+					manifold.points[j].normalImpulse = vc.points[j].normalImpulse;
+					manifold.points[j].tangentImpulse = vc.points[j].tangentImpulse;
+				}
+			}
+		}
+
+
+		public bool SolvePositionConstraints()
+		{
+			var minSeparation = 0.0f;
+
+			for (var i = 0; i < _count; ++i)
+			{
+				ContactPositionConstraint pc = _positionConstraints[i];
+
+				int indexA = pc.indexA;
+				int indexB = pc.indexB;
+				Vector2 localCenterA = pc.localCenterA;
+				float mA = pc.invMassA;
+				float iA = pc.invIA;
+				Vector2 localCenterB = pc.localCenterB;
+				float mB = pc.invMassB;
+				float iB = pc.invIB;
+				int pointCount = pc.pointCount;
+
+				Vector2 cA = _positions[indexA].c;
+				float aA = _positions[indexA].a;
+
+				Vector2 cB = _positions[indexB].c;
+				float aB = _positions[indexB].a;
+
+				// Solve normal constraints
+				for (var j = 0; j < pointCount; ++j)
+				{
+					var xfA = new Transform();
+					var xfB = new Transform();
+					xfA.q = Matrex.CreateRotation(aA);                   // Actually about twice as fast to use our own function
+					xfB.q = Matrex.CreateRotation(aB);                   // Actually about twice as fast to use our own function
+					xfA.p = cA - Vector2.Transform(localCenterA, xfA.q); // Common.Math.Mul(xfA.q, localCenterA);
+					xfB.p = cB - Vector2.Transform(localCenterB, xfB.q); // Common.Math.Mul(xfB.q, localCenterB);
+
+					PositionSolverManifold.Solve(pc, xfA, xfB, j, out Vector2 normal, out Vector2 point, out float separation);
```

Comment:
Also changed PositionSolverManifold into a static class with a static method with out params, instead of creating a new instance each time.

Reply:
And it's more readable. OK for me.